### PR TITLE
:umbrella: Add unit tests to ipynb conversion

### PR DESCRIPTION
**Pull request by FabienRoger**
Add tests for ipynb conversion

### CODE REVIEW

**Comment by FabienRoger** [comment]
I think the test I added cause an out-of-memory error. I don't know how to fix it.

File: tests/unit/scene/test_ipynb_conversion.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import pytest_check as check
+import json
+
+from opencodeblocks.scene.from_ipynb_conversion import ipynb_to_ipyg, is_title
+from opencodeblocks.scene.ipynb_conversion_constants import BLOCK_TYPE_TO_NAME
+
+
+class TestIpynbConversion:
+
+    """Testing function_parsing functions"""
```
Comment (MathisFederico) [inline tests/unit/scene/test_ipynb_conversion.py:13]: Wrong name, also look at clipboard test for naming conventions, they are different for tests.

```suggestion
    """Conversion from .ipynb"""
```

File: tests/unit/scene/test_ipynb_conversion.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    def test_empty_data(self, mocker: MockerFixture):
+        """Empty data should return empty document"""
```
Comment (MathisFederico) [inline tests/unit/scene/test_ipynb_conversion.py:16]:
```suggestion
        """should return empty graph for empty data."""
```

File: tests/unit/scene/test_ipynb_conversion.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    From ipynb conversion on a real notebooks should return
+    1. blocks and edges
+    2. the right amount of code blocks and edges
+    3. blocks and sockets with unique ids
+    4. edges with existing ids
+    5. code blocks that always have a source
+    6. markdown blocks that always have text
+    """
+    ipynb_data = load_json(file_path)
+    ipyg_data = ipynb_to_ipyg(ipynb_data)
```
Comment (MathisFederico) [inline tests/unit/scene/test_ipynb_conversion.py:62]: Those should be inputs for more modular behavior

File: tests/unit/scene/test_ipynb_conversion.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+class TestIpynbConversion:
+
+    """Testing function_parsing functions"""
+
+    def test_empty_data(self, mocker: MockerFixture):
+        """Empty data should return empty document"""
+        check.equal(ipynb_to_ipyg({}), {"blocks": [], "edges": []})
+
+    def test_empty_notebook_data(self, mocker: MockerFixture):
+        """The conversion for an empty notebook should be coherent"""
```
Comment (MathisFederico) [inline tests/unit/scene/test_ipynb_conversion.py:20]:
```suggestion
        """should return empty graph for empty notebook data."""
```

File: tests/unit/scene/test_ipynb_conversion.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        file_path = "./tests/assets/empty.ipynb"
+        real_notebook_is_coherent(file_path)
+
+    def test_complex_notebook_data(self, mocker: MockerFixture):
+        """The conversion for a usual notebook should be coherent"""
```
Comment (MathisFederico) [inline tests/unit/scene/test_ipynb_conversion.py:25]:
```suggestion
        """should return expected graph for usual notebook data."""
```

File: tests/unit/scene/test_ipynb_conversion.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        file_path = "./tests/assets/usual.ipynb"
+        real_notebook_is_coherent(file_path)
+
+    def test_complex_notebook_data(self, mocker: MockerFixture):
+        """The conversion for a complex notebook should be coherent"""
```
Comment (MathisFederico) [inline tests/unit/scene/test_ipynb_conversion.py:30]:
```suggestion
        """should return expected graph for complex notebook data."""
```

File: tests/unit/scene/test_ipynb_conversion.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        file_path = "./tests/assets/complex.ipynb"
+        real_notebook_is_coherent(file_path)
+
+    def test_is_title(self, mocker: MockerFixture):
+        """Should return True iff the given text can be used as a title for a block"""
```
Comment (MathisFederico) [inline tests/unit/scene/test_ipynb_conversion.py:35]:
```suggestion
        """should return True iff the given text can be used as a title for a block."""
```

File: tests/unit/scene/test_ipynb_conversion.py [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+def real_notebook_is_coherent(file_path: str):
+    """
+    From ipynb conversion on a real notebooks should return
+    1. blocks and edges
+    2. the right amount of code blocks and edges
+    3. blocks and sockets with unique ids
+    4. edges with existing ids
+    5. code blocks that always have a source
+    6. markdown blocks that always have text
```
Comment (MathisFederico) [inline tests/unit/scene/test_ipynb_conversion.py:59]: Do not comply with google docstring guidelines that we follow : https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings

Btw we should add that to CONTRIBUTING.md, you can do it if you want to.

File: tests/unit/scene/test_ipynb_conversion.py [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    """
+    ipynb_data = load_json(file_path)
+    ipyg_data = ipynb_to_ipyg(ipynb_data)
+
+    # 1.
```
Comment (MathisFederico) [inline tests/unit/scene/test_ipynb_conversion.py:64]: Use text instead of numbers.
```suggestion
    # Blocks and edges
```

**Review by MathisFederico — CHANGES_REQUESTED** [review]
Can't wait to merge that ! Tests are always welcome !

**Review by MathisFederico — APPROVED** [review]
Perfect !
Thank you a lot !
Tests are always very welcome !

**Pull request merged**